src/idl_gen_cpp.cpp

```diff
@@ -2083,9 +2083,15 @@ class CppGenerator : public BaseGenerator {
         if (!field.deprecated) {
           code_.SetValue("FIELD_NAME", Name(field));
           if (field.value.type.base_type == BASE_TYPE_STRING) {
-            code_ +=
-                "  auto {{FIELD_NAME}}__ = {{FIELD_NAME}} ? "
-                "_fbb.CreateString({{FIELD_NAME}}) : 0;";
+            if (!field.native_shared) {
+              code_ +=
+                  "  auto {{FIELD_NAME}}__ = {{FIELD_NAME}} ? "
```

Comment:
this line can be lifted outside of the if-then? In fact only CreateString / CreateSharedString needs to be inside of it.

Reply:
The `code_` is not a string but a `CodeWriter`. Each `+=` will implicitly append a line break at the end. In order to lift out `CreateString`, we need to either use a {{tag}}, or create a temporary string. I will switch to using a tag soon.

Reply:
Ahh sorry forgot about that.